lib/i18n/backend/base.rb

```diff
@@ -11,9 +11,12 @@ module Base
       # Accepts a list of paths to translation files. Loads translations from
       # plain Ruby (*.rb), YAML files (*.yml), or JSON files (*.json). See #load_rb, #load_yml, and #load_json
       # for details.
-      def load_translations(*filenames)
+      def load_translations(*filenames, &block)
         filenames = I18n.load_path if filenames.empty?
-        filenames.flatten.each { |filename| load_file(filename) }
+        filenames.flatten.each do |filename|
+          loaded_translations = load_file(filename)
+          block.call(filename, loaded_translations) if block_given?
```

Comment:
It might be nicer to use an implicit block here (and should be faster, because Ruby won't have to create an object):
```suggestion
          yield filename, loaded_translations if block_given?
```